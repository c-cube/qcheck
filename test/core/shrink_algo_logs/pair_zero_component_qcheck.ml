open QCheck

let pair_one_zero =
  Test.make ~name:"pairs have a zero component" (pair int int)
    (Log.shrinks Print.(pair int int) @@ fun (i,j) -> i=0 || j=0)
;;
try Test.check_exn pair_one_zero with Test.Test_fail _ -> ()
