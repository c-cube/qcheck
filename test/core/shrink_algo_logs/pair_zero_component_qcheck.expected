fails (-4202639126704614816, 2724675603984413065)
fails (-2101319563352307408, 2724675603984413065)
fails (-1050659781676153704, 2724675603984413065)
fails (-525329890838076852, 2724675603984413065)
fails (-262664945419038426, 2724675603984413065)
fails (-131332472709519213, 2724675603984413065)
fails (-65666236354759607, 2724675603984413065)
fails (-32833118177379804, 2724675603984413065)
fails (-16416559088689902, 2724675603984413065)
fails (-8208279544344951, 2724675603984413065)
fails (-4104139772172476, 2724675603984413065)
fails (-2052069886086238, 2724675603984413065)
fails (-1026034943043119, 2724675603984413065)
fails (-513017471521560, 2724675603984413065)
fails (-256508735760780, 2724675603984413065)
fails (-128254367880390, 2724675603984413065)
fails (-64127183940195, 2724675603984413065)
fails (-32063591970098, 2724675603984413065)
fails (-16031795985049, 2724675603984413065)
fails (-8015897992525, 2724675603984413065)
fails (-4007948996263, 2724675603984413065)
fails (-2003974498132, 2724675603984413065)
fails (-1001987249066, 2724675603984413065)
fails (-500993624533, 2724675603984413065)
fails (-250496812267, 2724675603984413065)
fails (-125248406134, 2724675603984413065)
fails (-62624203067, 2724675603984413065)
fails (-31312101534, 2724675603984413065)
fails (-15656050767, 2724675603984413065)
fails (-7828025384, 2724675603984413065)
fails (-3914012692, 2724675603984413065)
fails (-1957006346, 2724675603984413065)
fails (-978503173, 2724675603984413065)
fails (-489251587, 2724675603984413065)
fails (-244625794, 2724675603984413065)
fails (-122312897, 2724675603984413065)
fails (-61156449, 2724675603984413065)
fails (-30578225, 2724675603984413065)
fails (-15289113, 2724675603984413065)
fails (-7644557, 2724675603984413065)
fails (-3822279, 2724675603984413065)
fails (-1911140, 2724675603984413065)
fails (-955570, 2724675603984413065)
fails (-477785, 2724675603984413065)
fails (-238893, 2724675603984413065)
fails (-119447, 2724675603984413065)
fails (-59724, 2724675603984413065)
fails (-29862, 2724675603984413065)
fails (-14931, 2724675603984413065)
fails (-7466, 2724675603984413065)
fails (-3733, 2724675603984413065)
fails (-1867, 2724675603984413065)
fails (-934, 2724675603984413065)
fails (-467, 2724675603984413065)
fails (-234, 2724675603984413065)
fails (-117, 2724675603984413065)
fails (-59, 2724675603984413065)
fails (-30, 2724675603984413065)
fails (-15, 2724675603984413065)
fails (-8, 2724675603984413065)
fails (-4, 2724675603984413065)
fails (-2, 2724675603984413065)
fails (-1, 2724675603984413065)
holds (0, 2724675603984413065)
fails (-1, 1362337801992206533)
holds (0, 1362337801992206533)
fails (-1, 681168900996103267)
holds (0, 681168900996103267)
fails (-1, 340584450498051634)
holds (0, 340584450498051634)
fails (-1, 170292225249025817)
holds (0, 170292225249025817)
fails (-1, 85146112624512909)
holds (0, 85146112624512909)
fails (-1, 42573056312256455)
holds (0, 42573056312256455)
fails (-1, 21286528156128228)
holds (0, 21286528156128228)
fails (-1, 10643264078064114)
holds (0, 10643264078064114)
fails (-1, 5321632039032057)
holds (0, 5321632039032057)
fails (-1, 2660816019516029)
holds (0, 2660816019516029)
fails (-1, 1330408009758015)
holds (0, 1330408009758015)
fails (-1, 665204004879008)
holds (0, 665204004879008)
fails (-1, 332602002439504)
holds (0, 332602002439504)
fails (-1, 166301001219752)
holds (0, 166301001219752)
fails (-1, 83150500609876)
holds (0, 83150500609876)
fails (-1, 41575250304938)
holds (0, 41575250304938)
fails (-1, 20787625152469)
holds (0, 20787625152469)
fails (-1, 10393812576235)
holds (0, 10393812576235)
fails (-1, 5196906288118)
holds (0, 5196906288118)
fails (-1, 2598453144059)
holds (0, 2598453144059)
fails (-1, 1299226572030)
holds (0, 1299226572030)
fails (-1, 649613286015)
holds (0, 649613286015)
fails (-1, 324806643008)
holds (0, 324806643008)
fails (-1, 162403321504)
holds (0, 162403321504)
fails (-1, 81201660752)
holds (0, 81201660752)
fails (-1, 40600830376)
holds (0, 40600830376)
fails (-1, 20300415188)
holds (0, 20300415188)
fails (-1, 10150207594)
holds (0, 10150207594)
fails (-1, 5075103797)
holds (0, 5075103797)
fails (-1, 2537551899)
holds (0, 2537551899)
fails (-1, 1268775950)
holds (0, 1268775950)
fails (-1, 634387975)
holds (0, 634387975)
fails (-1, 317193988)
holds (0, 317193988)
fails (-1, 158596994)
holds (0, 158596994)
fails (-1, 79298497)
holds (0, 79298497)
fails (-1, 39649249)
holds (0, 39649249)
fails (-1, 19824625)
holds (0, 19824625)
fails (-1, 9912313)
holds (0, 9912313)
fails (-1, 4956157)
holds (0, 4956157)
fails (-1, 2478079)
holds (0, 2478079)
fails (-1, 1239040)
holds (0, 1239040)
fails (-1, 619520)
holds (0, 619520)
fails (-1, 309760)
holds (0, 309760)
fails (-1, 154880)
holds (0, 154880)
fails (-1, 77440)
holds (0, 77440)
fails (-1, 38720)
holds (0, 38720)
fails (-1, 19360)
holds (0, 19360)
fails (-1, 9680)
holds (0, 9680)
fails (-1, 4840)
holds (0, 4840)
fails (-1, 2420)
holds (0, 2420)
fails (-1, 1210)
holds (0, 1210)
fails (-1, 605)
holds (0, 605)
fails (-1, 303)
holds (0, 303)
fails (-1, 152)
holds (0, 152)
fails (-1, 76)
holds (0, 76)
fails (-1, 38)
holds (0, 38)
fails (-1, 19)
holds (0, 19)
fails (-1, 10)
holds (0, 10)
fails (-1, 5)
holds (0, 5)
fails (-1, 3)
holds (0, 3)
fails (-1, 2)
holds (0, 2)
fails (-1, 1)
holds (0, 1)
holds (-1, 0)
